From db2b3dfd0789cd433e5caf9d18ae72b181712bbf Mon Sep 17 00:00:00 2001
From: Rene Kuettner <rene@bitkanal.net>
Date: Fri, 28 Jun 2013 14:24:22 +0200
Subject: [PATCH] Fix sequence point

---
 05/include/tiny_vec.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/05/include/tiny_vec.h b/05/include/tiny_vec.h
index 4a17b47..641da61 100644
--- a/05/include/tiny_vec.h
+++ b/05/include/tiny_vec.h
@@ -815,7 +815,8 @@ T tiny_vec<T,N>::sqr_length() const
 	iterator it= (T*)begin();
 	T l = 0;
 	while(it != end())
-		l += (*it)*(*it++);
+		l += (*it)*(*it);
+		++it;
 
 	return l;
 }
-- 
1.8.3.1

