diff --git a/03_graph_prog_orig/include/abstract_scene.h b/03_graph_prog/include/abstract_scene.h
index 7096c6b..972f40b 100644
--- a/03_graph_prog_orig/include/abstract_scene.h
+++ b/03_graph_prog/include/abstract_scene.h
@@ -18,6 +18,8 @@ using namespace std;
 class abstract_scene
 {
 public:
+	virtual ~abstract_scene();
+
 	// Render the scene
 	virtual void render() = 0;
 
diff --git a/03_graph_prog_orig/src/abstract_scene.cpp b/03_graph_prog/src/abstract_scene.cpp
index f8b0f0c..26c388b 100644
--- a/03_graph_prog_orig/src/abstract_scene.cpp
+++ b/03_graph_prog/src/abstract_scene.cpp
@@ -1,6 +1,13 @@
 #include "abstract_scene.h"
 
 
+// required to avoid undefined behaviour when deleting a
+// pointer of type abstract_scene
+abstract_scene::~abstract_scene()
+{
+}
+
+
 // Put a text into the stream "stream" which will be displayed
 // in the main window
 void abstract_scene::set_text(stringstream &stream)
